src/main/kotlin/com/epam/ahnl/service/mapper/StudentMapper.kt

```diff
@@ -0,0 +1,19 @@
+package com.epam.ahnl.service.mapper
+
+import com.epam.ahnl.model.entity.Student
+import com.epam.ahnl.service.dto.StudentDto
+import com.epam.ahnl.service.dto.StudentView
+import kotlin.reflect.full.memberProperties
+
+fun Student.toStudentView() = StudentView(
+        id ?: "empty",
+        "$firstName $lastName",
+        birthDate,
+        faculty,
+        address
+)
+
+fun StudentDto.toStudent() = with(::Student) {
+    val propertiesByName = StudentDto::class.memberProperties.associateBy { it.name }
```

Comment:
So, 1) if we remove 1 dto class, 1 extension function will be gone
2) we were taught to avoid using extension function often, cause code becomes confusing.
3) I'd create mapper class, to decompose logic. SOLID and so on. You used reflection for mapping. Honestly had to google reflection in kotlin, but it's good experience. And my question is what if the structure dto and entity will be different. Wil lit work ?

Reply:
Mappers were implemented like extension functions.